x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/optimizer/OptimizerRules.java

```diff
@@ -1137,6 +1142,51 @@ protected Expression rule(Expression e) {
         }
     }
 
+    public static final class BubbleUpNegations extends OptimizerExpressionRule {
+
+        public BubbleUpNegations() {
+            super(TransformDirection.DOWN);
+        }
+
+        @Override
+        protected Expression rule(Expression e) {
+            if (e instanceof Neg) {
+                Expression child = ((Neg) e).field();
+                if (child instanceof ArithmeticOperation) { // cancels out NEG - MUL/DIV - NEG - x ==> MUL/DIV - x
+                    ArithmeticOperation operation = (ArithmeticOperation) child;
+                    if (isMulOrDiv(operation)) {
+                        if (operation.left() instanceof Neg) {
+                            return operation.replaceChildren(List.of(((Neg) operation.left()).field(), operation.right()));
+                        }
+                        if (operation.right() instanceof Neg) {
+                            return operation.replaceChildren(List.of(operation.left(), ((Neg) operation.right()).field()));
+                        }
+                    }
+                } else if (child instanceof Neg) { // cancels out NEG - NEG - x ==> x
+                    return ((Neg) child).field();
+                }
+            } else if (e instanceof ArithmeticOperation) { // pulls up MUL/DIV - NEG - x ==> NEG - MUL/DIV - x
+                ArithmeticOperation operation = (ArithmeticOperation) e;
+                if (isMulOrDiv(operation)) {
+                    if (operation.left() instanceof Neg) {
+                        Neg neg = (Neg) operation.left();
+                        return new Neg(operation.source(), operation.replaceChildren(List.of(neg.field(), operation.right())));
+                    }
+                    if (operation.right() instanceof Neg) {
+                        Neg neg = (Neg) operation.right();
+                        return new Neg(operation.source(), operation.replaceChildren(List.of(operation.left(), neg.field())));
+                    }
+                }
+            }
+            return e;
+        }
+
+        public static boolean isMulOrDiv(ArithmeticOperation operation) {
```

Comment:
`public` as extracted from #66022, where it's being reused in another rule.
Will make eventually `protected`, however.